src/commons/BarChartData.js

```diff
@@ -0,0 +1,93 @@
+class BarChartData {
+  /**
+   * Validate behavior of graph according to id indicator
+   *
+   * @param {Integer} idIndicator id indicator to be validated
+   */
+  static validateIndicator(idIndicator) {
+    let validIndicator = false;
+    let indicatorNamePositive = 'No definido';
+    let indicatorNameNegative = 'No definido';
+
+    switch (idIndicator) {
+      case 12:
+        validIndicator = true;
+        indicatorNamePositive = 'Superficie del área protegida dentro del bloque';
+        indicatorNameNegative = 'Superficie del área protegida fuera del bloque';
+        break;
+      case 18:
+        validIndicator = true;
+        indicatorNamePositive = 'Superficie del área de reserva de ley 2da de 1959 dentro del bloque';
+        indicatorNameNegative = 'Superficie del área de reserva de ley 2da de 1959 fuera del bloque';
+        break;
+      case 20:
+        validIndicator = true;
+        indicatorNamePositive = 'Superficie del área del territorio étnico dentro del bloque';
+        indicatorNameNegative = 'Superficie del área del territorio étnico fuera del bloque';
+        break;
+      case 22:
+        validIndicator = true;
+        indicatorNamePositive = 'Superficie del área de la ZRC dentro del bloque';
+        indicatorNameNegative = 'Superficie del área de la ZRC fuera del bloque';
+        break;
+      case 24:
+        validIndicator = true;
+        indicatorNamePositive = 'Superficie del área del ecosistema estrategico dentro del bloque';
+        indicatorNameNegative = 'Superficie del área del ecosistema estrategico fuera del bloque';
+        break;
+      default:
+        validIndicator = false;
+        break;
+    }
+
+    return { validIndicator, indicatorNamePositive, indicatorNameNegative };
+  }
+  /**
+   * Construct an array to be inserted in the transformed graph data
+   *
+   * @param {Array} rawData
+   * @param {String} valueDescription
+   */
+
+  static processOneRow(rawData, valueDescription) {
+    const row = [];
+    const rowComputed = [];
+    row.push(valueDescription);
+    rawData.forEach((item) => {
+      if (item.value_description === valueDescription) {
+        row.push(parseFloat(item.indicator_value, 10));
+      }
+    });
+    const desc = row[0];
+    const value = row[1];
+    const valueTransf = (((100 - row[2]) * row[1]) / row[2]);
```

Comment:
The problem with this is that you are assuming that values will always be sorted and the percentage will come later than the area number.

row should be an object where you can be sure which value is which. In that case you need to map somewhere the ids pairs (area value, percentage)